internal/pkg/scorecard/helpers.go

```diff
@@ -35,3 +40,92 @@ func ResultsCumulative(results []TestResult) (earned, max int) {
 	}
 	return earned, max
 }
+
+// CalculateStates updates the state fields of the JSONOut based on the TestResults in out.Tests
+func CalculateStates(out *scapiv1alpha1.ScorecardResult) {
+	out.Error = 0
+	out.Pass = 0
+	out.PartialPass = 0
+	out.Fail = 0
+	out.TotalTests = 0
+	for _, test := range out.Tests {
+		out.TotalTests++
+		switch test.State {
+		case scapiv1alpha1.ErrorState:
+			out.Error++
+		case scapiv1alpha1.PassState:
+			out.Pass++
+		case scapiv1alpha1.PartialPassState:
+			out.PartialPass++
+		case scapiv1alpha1.FailState:
+			out.Fail++
+		}
+	}
+}
+
+// TestSuitesToScorecardTest takes an array of test suites and generates a v1alpha1 ScorecardTest object with the
+// provided name, description, and log
+func TestSuitesToScorecardTest(suites []*TestSuite, name, description, log string) *scapiv1alpha1.ScorecardTest {
+	test := &scapiv1alpha1.ScorecardTest{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Scorecard",
+			APIVersion: "scorecard/v1alpha1",
+		},
+		Spec: &scapiv1alpha1.ScorecardTestSpec{
+			Name:        name,
+			Description: description,
+		},
+		Status: &scapiv1alpha1.ScorecardTestResult{
+			Log: log,
+		},
+	}
+	scorecardResults := []scapiv1alpha1.ScorecardResult{}
+	for _, suite := range suites {
+		scorecardResult := scapiv1alpha1.ScorecardResult{}
+		results := []*scapiv1alpha1.JSONTestResult{}
+		for _, testResult := range suite.TestResults {
+			results = append(results, TestResultToJSONTestResult(testResult))
+		}
+		scorecardResult.Tests = results
+		scorecardResult.TotalScore = suite.TotalScore()
+		CalculateStates(&scorecardResult)
+		scorecardResults = append(scorecardResults, scorecardResult)
+	}
+	test.Status.Results = scorecardResults
+	return test
+}
+
+// TestResultToJSONTestResult is a helper function for converting from the TestResult type to the JSONTestResult type
+func TestResultToJSONTestResult(tr *TestResult) *scapiv1alpha1.JSONTestResult {
```

Comment:
Do you really need a pointer to TestResult?

Reply:
Not really. But it's more efficient to pass structs as pointers instead of the entire struct

Reply:
the efficiency gain is so small while being clear what you're doing. In this case, it is doing nothing to TestResult. It seems much more important to me to make it clear that we are not messing with test result just using them to make the v1alpha1JSONTestResult.